### PR TITLE
fix: No support for python greater than 3.9

### PR DESCRIPTION
**Pull request by askalski85**
Introduce proper version check for python 3.10+

Sample output:
```
---
Searching for Python in Path: /Users/skalski/.sdkman/candidates/java/current/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/skalski/Library/Python/3.8/bin
- command python not found.
- command python3 [3.4.1] is found but the minimum required version is '3.5'.
- command python2 [2.2] is found but the minimum required version is '2.6'.
! Suitable Python interpreter not found.
---
Searching for Python in Path: ...
- command python not found.
- command python3 [3.4.1] is found but the minimum required version is '3.5'.
+ command python2 [2.6] is found and matches the minimum required version - Success!
---
Searching for Python in Path:...
- command python not found.
+ command python3 [3.10.1] is found and matches the minimum required version - Success!
```

### CODE REVIEW

**Comment by askalski85** [comment]
fyi: @Kami @czerwingithub

**Comment by ArkadiuszSkalski-TomTom** [comment]
fyi @ArthurKamalov

**Comment by askalski85** [comment]
fyi: @ArthurKamalov

**Comment by askalski85** [comment]
ping

**Comment by ArthurKamalov** [comment]
@askalski85 Hey! Thanks for your contribution, I'll review it as soon a possible.

**Comment by Kami** [comment]
@ArthurKamalov IIRC, we had a similar bug in the past, would be good if we could add some direct unit tests for this script (IIRC, we already have some end to end test cases, but I believe we don't cover all the edge cases for Python versions).

**Comment by Kami** [comment]
I noticed we have similar check in pre install and post install script (https://github.com/scalyr/scalyr-agent-2/blob/master/installer/scripts/preinstall.sh#L24, ). I assume we need to update both places?

**Comment by askalski85** [comment]
> I noticed we have similar check in pre install and post install script (https://github.com/scalyr/scalyr-agent-2/blob/master/installer/scripts/preinstall.sh#L24, https://github.com/scalyr/scalyr-agent-2/blob/master/installer/scripts/postinstall.sh). I assume we need to update both places?

oh its possible. Ill take a look into it. It should be possible to extract it to some common place maybe :-)

File: installer/scripts/preinstall.sh [hunk 1]
```diff
@@ -18,44 +18,68 @@
 # In this cases script have to exit with an error.
 # This is important because all agent scripts rely on '/usr/bin/env python' command.
 
+# Requirements
+MIN_REQ_PYTHON2_VERSION="2.6"
```
[the remark this answers is not in the document]
Comment (askalski85) [reply installer/scripts/preinstall.sh]: done

**Comment by ArthurKamalov** [comment]
> > I noticed we have similar check in pre install and post install script (https://github.com/scalyr/scalyr-agent-2/blob/master/installer/scripts/preinstall.sh#L24, https://github.com/scalyr/scalyr-agent-2/blob/master/installer/scripts/postinstall.sh). I assume we need to update both places?
> 
> oh its possible. Ill take a look into it. It should be possible to extract it to some common place maybe :-)

Unfortunately, we can not access any of agent's files (including `/usr/share/scalyr-agent-2/bin/check-python.sh`) from  `preinstall` script because those files are not copied yet. That's why we have that logic duplication. 

As a workaround, we can use pre and post install scripts as templates and paste logic from the `check-python.sh` into them during the build. @Kami WDYT?

**Comment by askalski85** [comment]
Lets have the pre install script including the python logic. Its not doing anything except checking the python anyway. The post install can call the logic from pre install script to avoid code duplication. At post install time all the scripts are available. Ill try to propose an update.

I checked briefly the templates I see you can inject variables easily:
`--template-scripts --before-install x --template-value aa=22`, yet for a more complex long bash script this will not be nice.

**Comment by askalski85** [comment]
> > Lets have the pre install script including the python logic. Its not doing anything except checking the python anyway. The post install can call the logic from pre install script to avoid code duplication. At post install time all the scripts are available. Ill try to propose an update.
> > I checked briefly the templates I see you can inject variables easily: `--template-scripts --before-install x --template-value aa=22`, yet for a more complex long bash script this will not be nice.
> 
> > Lets have the pre install script including the python logic. Its not doing anything except checking the python anyway. The post install can call the logic from pre install script to avoid code duplication. At post install time all the scripts are available. Ill try to propose an update.
> > I checked briefly the templates I see you can inject variables easily: `--template-scripts --before-install x --template-value aa=22`, yet for a more complex long bash script this will not be nice.
> 
> I am also about to push a similar approach, I've also made additional changes to the build process and tests in order to make everything work. I'll do it in another PR since I am not sure how to push changes in your 😅 It also seems that our circleci tests do not trigger from forks.

We can change target of my PR to some other branch and you can take it from there if you like.

**Comment by ArthurKamalov** [comment]
> We can change target of my PR to some other branch and you can take it from there if you like.

@askalski85 Could you please change it to `fix_install_scrits_for_python3_10`.

**Comment by askalski85** [comment]
> > We can change target of my PR to some other branch and you can take it from there if you like.
> 
> @askalski85 Could you please change it to `fix_install_scrits_for_python3_10`.

Done! Thanks for taking care of this @ArthurKamalov!

**Comment by askalski85** [comment]
> > > We can change target of my PR to some other branch and you can take it from there if you like.
> > 
> > 
> > @askalski85 Could you please change it to `fix_install_scrits_for_python3_10`.
> 
> Done! Thanks for taking care of this @ArthurKamalov!

I was rather expecting this is a fresh branch, now I did hit some conflicts. I need to find some time to resolve them, whereas all ubuntu22.04 users are blocked. :(

**Comment by askalski85** [comment]
@ArthurKamalov @Kami decided to open a new PR https://github.com/scalyr/scalyr-agent-2/pull/867/files which does not solve the code duplication issue, but unblocks the python >3.9 check and all ubuntu 22.04 users.

**Comment by askalski85** [comment]
the fix for python3.9 merged with https://github.com/scalyr/scalyr-agent-2/commit/96363915a271a2c67adf61e9c5fed9a883694334

We can abandon this for now.

**Comment by Kami** [comment]
@askalski85 @rodrigodelmonte v2.1.30 release with this bug fix is now out - https://github.com/scalyr/scalyr-agent-2/releases/tag/v2.1.30. Please let us know when you get a chance to confirm the fix with v2.1.3.0.

Thanks for reporting this issue and thanks for contribution the patch.
